ensawards.org/src/components/atoms/badges/MatrixBadge.tsx

```diff
@@ -0,0 +1,90 @@
+import { GenericTooltip } from "@/components/atoms/GenericTooltip.tsx";
+import { ResolveAndDisplayIdentity } from "@/components/atoms/identity";
+import { TooltipProvider } from "@/components/ui/tooltip.tsx";
+import { type BenchmarkResult, type BenchmarkVerifier } from "@/types/apps.ts";
+import { getENSNodeUrl } from "@/utils/env";
+import { ENSNodeProvider, createConfig } from "@ensnode/ensnode-react";
+import { ENSNamespaceIds, ENSNodeClient, type UnresolvedIdentity } from "@ensnode/ensnode-sdk";
+import { Check, CheckCheck, X as XIcon } from "lucide-react";
+import React from "react";
+import { cn } from "../../../utils/tailwindClassConcatenation";
+
+export interface MatrixBadgeProps {
+  text: string;
+  className: string;
+  result: BenchmarkResult;
+  verification?: BenchmarkVerifier;
+}
+
+const TooltipContent = ({ verification }: { verification?: BenchmarkVerifier }) => {
+  if (!verification) {
+    return <p className="max-w-[275px]">This benchmark has not been verified yet.</p>;
+  }
+
+  const identity: UnresolvedIdentity = {
+    chainId: verification.chainId,
+    address: verification.address,
+    resolutionStatus: "unresolved",
+  };
+
+  const verifiedDate = new Date(verification.verifiedAt * 1000).toLocaleDateString("en-US", {
+    year: "numeric",
+    month: "long",
+    day: "numeric",
+  });
+
+  return (
+    <div className="max-w-[275px] flex flex-col gap-2">
+      <div className="text-sm">
+        <ResolveAndDisplayIdentity
+          identity={identity}
+          namespaceId={ENSNamespaceIds.Mainnet}
```

Comment:
Need to update this to be dynamic but facing some issues due to the types used in the dependent component

Reply:
Resolved